### PR TITLE
[Network] `az network vnet-gateway create`: Add parameter `--edge-zone-vnet-id` for local gateway

### PR DESCRIPTION
**Pull request by tguxia**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
network

**Review by necusjz — CHANGES_REQUESTED** [review]
The title of this PR is a little bit rough, could you please refine it? As well as the linter/style checks.

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 1]
```diff
@@ -1935,7 +1936,7 @@ def load_arguments(self, _):
         c.argument('sku', vnet_gateway_sku_type)
         c.argument('vpn_type', vnet_gateway_routing_type)
         c.argument('bgp_peering_address', arg_group='BGP Peering', help='IP address to use for BGP peering.')
-        c.argument('public_ip_address', options_list=['--public-ip-addresses'], nargs='+', help='Specify a single public IP (name or ID) for an active-standby gateway. Specify two space-separated public IPs for an active-active gateway.', completer=get_resource_name_completion_list('Microsoft.Network/publicIPAddresses'))
+        c.argument('public_ip_address', required=False, options_list=['--public-ip-addresses'], nargs='+', help='Specify a single public IP (name or ID) for an active-standby gateway. Specify two space-separated public IPs for an active-active gateway.', completer=get_resource_name_completion_list('Microsoft.Network/publicIPAddresses'))
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:1939]: We don't need to explicitly declare "required or not". It's determined by the function signature in _custom.py_ (positional or keyword).

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 2]
```diff
@@ -1965,6 +1966,7 @@ def load_arguments(self, _):
         c.argument('virtual_network', options_list='--vnet', help=vnet_help)
         c.argument('vpn_gateway_generation', arg_type=get_enum_type(['Generation1', 'Generation2']), min_api='2019-07-01', help='The generation for the virtual network gateway. vpn_gateway_generation should not be provided if gateway_type is not Vpn.')
         c.argument('edge_zone', edge_zone, min_api='2021-02-01')
+        c.argument('VNetExtendedLocationResourceId', VNetExtendedLocationResourceId, min_api='2021-02-01')
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:1969]: "min_api='2021-02-01" seems redundant.

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py [hunk 3]
```diff
@@ -6222,6 +6222,22 @@ def test_network_vnet_gateway_edge_zone(self, resource_group):
...
+    @ResourceGroupPreparer(name_prefix='test_network_vnet_gateway_local_gateway', location='eastus2euap')
+    def test_network_vnet_gateway_local_gateway(self, resource_group):
+
+        self.kwargs.update({
+            'rg': resource_group,
+            'vnet': 'vnet',
+            'gateway_type' : 'LocalGateway',
+            'edge_name': 'microsoftrrdclab3',
+            'edge_zone_vnet_id' : 'edge_zone_vnet_id'
+        })
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py:6234]:
```suggestion
        self.kwargs.update({
            'rg': resource_group,
            'vnet': 'vnet',
            'gateway_type' : 'LocalGateway',
            'edge_name': 'microsoftrrdclab3',
        })
```
It's useless as we declare it later.

**Pull request merged**